ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,99 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+--jit.opt.start(1)
+--jit.opt.start("loopunroll=10")
+function BB_mt.__index:writeBMP(filename)
+    local function write_uint32(of, data)
+        of:write(string.char(band(data, 255)))
+        data = math.floor(data / 256)
+        of:write(string.char(band(data, 255)))
+        data = math.floor(data / 256)
+        of:write(string.char(band(data, 255)))
+        data = math.floor(data / 256)
+        of:write(string.char(data))
+    end
+
+    local output_channels = 3
+    local w, h = self:getWidth(), self:getHeight()
+    local stride = w * output_channels
+
+    local of, err = io.open(filename, "wb")
+    if err ~= nil then
+        return err
+    end
+
+    local bbdump
+    local data
+    if self:getType() == TYPE_BBRGB24 then
```

Comment:
You should be able to use this on BBRGB32, too, you'll just have to keep track of two different `pos` later on: the input one (using the source bb's stride), and the output one (using the output bb's stride).

Reply:
OK, I can do it. But how should I test it. Although I know how to do it, I don't feel comfortable just to program it without a test.
Is there a switch on the emulator for using RGB32?

Reply:
Both Android and the emulator should be running on an RGB32 screen buffer, IIRC, so running that from the screen BB like the Screenshotter does instead of an image file should do the trick?

Reply:
But, again, I'm thinking in terms of Screen buffers: this may be a terrible idea on actual RGBA images, since you wouldn't be doing alpha-blending.

TL;DR: Eh, probably don't bother ;p.